cloud-sql/mysql/Startup.cs

```diff
@@ -55,152 +54,157 @@ public void ConfigureServices(IServiceCollection services)
                     options.Version = "Test";
                 });
             }
-            services.AddScoped(typeof(DbConnection), (IServiceProvider) =>
-                InitializeDatabase());
+            services.AddScoped(typeof(MySqlConnectionStringBuilder),
+                (IServiceProvider) => new MySqlConnect().GetMySqlConnectionString());
+            services.AddScoped<MySqlConnect>();

```

Comment:
I think you can do with registering only the connection string. This extra registration is only there so you can call the method that adds the retry policy, but that doesn't need to be injected. It can be an extension method on the connection so that you can call `connection.OpenWithRetry()` or something similar. (See my other comment on the retry bits).

Reply:
Done.